src/components/MyNdla/CopyFolderModal.tsx

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright (c) 2022-present, NDLA.
+ *
+ * This source code is licensed under the GPLv3 license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+import { ReactNode, useCallback, useContext, useMemo, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { ListResource } from '@ndla/ui';
+import {
+  ModalBody,
+  ModalCloseButton,
+  ModalHeader,
+  ModalTitle,
+  Modal,
+  ModalTrigger,
+  ModalContent,
+} from '@ndla/modal';
+import { AuthContext } from '../AuthenticationContext';
+import { GQLFolder } from '../../graphqlTypes';
+import LoginModalContent from './LoginModalContent';
+import CopyFolder, { baseSharedFolder } from './CopyFolder';
+
+interface Props {
+  folder: GQLFolder;
+  children: ReactNode;
+}
+
+const CopyFolderModal = ({ folder, children }: Props) => {
+  const [open, setOpen] = useState(false);
+  const { t } = useTranslation();
+  const { authenticated } = useContext(AuthContext);
+
+  const close = useCallback(() => setOpen(false), []);
+
+  const sharedFolder = useMemo(() => baseSharedFolder(t), [t]);
+
+  return (
+    <Modal open={open} onOpenChange={setOpen}>
+      <ModalTrigger>{children}</ModalTrigger>
+      {authenticated ? (
+        <ModalContent>
+          <ModalHeader>
+            <ModalTitle>{t('myNdla.resource.copyToMyNdla')}</ModalTitle>
+            <ModalCloseButton title={t('modal.closeModal')} />
+          </ModalHeader>
+          <ModalBody>
+            <CopyFolder folder={folder} onClose={close} />
+          </ModalBody>
+        </ModalContent>
+      ) : (
+        <LoginModalContent
+          title={t('myNdla.loginCopyFolderPitch')}
+          content={
+            folder && (
+              <ListResource
+                id={folder.id.toString()}
+                link={`/folder/${folder.id}`}
+                title={folder.name ?? ''}
+                resourceImage={{
+                  src: '',
+                  alt: '',
+                }}
+                resourceTypes={[sharedFolder]}
+              />
```

Comment:
Hadde det gitt mer mening å bruke `Folder`-komponenten her?

Reply:
Ja, det syntes jeg egentlig var en god ide! Ifølge designet så ligner det mer på `ListResource`-komponenten, burde det avklares med NDLA hvis vi heller går for `Folder`-komponenten? @gunnarvelle har kanskje innspill

Isåfall vil det se ut ca som dette:
![Skjermbilde 2023-08-25 kl  10 21 33](https://github.com/NDLANO/ndla-frontend/assets/26788204/f107dff8-60ec-44e1-9bce-f794992546b8)

Reply:
Tenker vi kan bruke mappe-komponent. Det er tross alt en mappe som kopieres.